lib/rspec/core/example_group.rb

```diff
@@ -557,11 +556,17 @@ def self.run_examples(reporter)
           instance = new(example.inspect_output)
           set_ivars(instance, before_context_ivars)
           succeeded = example.run(instance, reporter)
-          RSpec.world.wants_to_quit = true if fail_fast? && !succeeded
+          if !succeeded && fail_fast?
+            RSpec.world.wants_to_quit = true if reporter.fail_fast_limit_met?
+          end
```

Comment:
I'd still like to see the two `if` statements here combined into one.  Nesting them seems messier and unnecessary to me.